### PR TITLE
MLflow UI sync

### PR DESCRIPTION
**Pull request by hubertzub-db**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/hubertzub-db/mlflow/pull/10163?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10163/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10163
```

</p>
</details>

### Related Issues/PRs

 #xxx

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [X] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@hubertzub-db Thanks for filing this PR! In OSS, we're still using node 16. Could this be an issue?

**Comment by hubertzub-db** [comment]
@harupy universe still uses node `16.18.0` so it should be ok 🙂

**Comment by harupy** [comment]
Got it :)

File: mlflow/server/js/package.json [hunk 1]
```diff
@@ -24,7 +24,9 @@
     "extract-i18n:check": "echo '\n*******************\n`yarn extract-i18n:check` command is now deprecated. Please use `yarn i18n:check` instead.'; exit 1",
     "extract-i18n": "echo '\n*******************\n`yarn extract-i18n` command is now deprecated. Please use `yarn i18n` instead.'; exit 1",
     "check-all": "yarn gocx-codegen:check && yarn type-check && yarn lint && yarn i18n:check",
-    "fix-all": "yarn gocx-codegen && yarn i18n && yarn lint:fix && yarn type-check",
+    "fix-all": "yarn gocx-codegen && yarn graphql-codegen && yarn i18n && yarn lint:fix && yarn type-check",
+    "graphql-codegen": "yarn graphql-codegen:clean && graphql-codegen --config ./src/graphql/graphql-codegen.ts && eslint src/graphql/__generated__/*.ts --fix",
```
Comment (xanderwebs) [inline mlflow/server/js/package.json:28]: Hmm do these actually work? We don't have any graphql in oss yet right? maybe we should wait to introduce?

Comment (hubertzub-db) [reply mlflow/server/js/package.json:28]: Yeah, I'm ok with removing those for now

**Review by xanderwebs — APPROVED** [review]
I only checked the common + shared files, but everything lgtm; no unwanted crossovers 😉

**Review by sunishsheth2009 — APPROVED** [review]
Overall seems good. glanced through the code. Nothing seems out of the ordinary. :)

**Comment by harupy** [comment]
@hubertzub-db Can you update `yarn.lock` to fix https://github.com/mlflow/mlflow/actions/runs/6655605769/job/18086232359?pr=10163?

**Review by harupy — APPROVED** [review]
LGTM!

File: mlflow/server/js/.eslintrc.js [hunk 2]
```diff
@@ -471,9 +466,13 @@ module.exports = {
...
         'plugin:@typescript-eslint/recommended',
       ],
-      plugins: ['prettier', 'react', '@typescript-eslint', '@emotion'],
+      plugins: ['prettier', '@typescript-eslint', '@emotion'],
       parser: '@typescript-eslint/parser',
 
+      // BEGIN-EDGE
+      // While we're waiting to have code sharing here, the following rules
+      // are plucked from js/packages/config-eslint/presets/createConfig.js
+      // END-EDGE
```
Comment (harupy) [inline mlflow/server/js/.eslintrc.js:475]: Shouldn't this be exported?

Comment (hubertzub-db) [reply mlflow/server/js/.eslintrc.js:475]: hmm those blocks seem to be left in [`.eslintrc.oss.js`](https://src.dev.databricks.com/databricks/universe/-/blob/mlflow/web/js/oss/.eslintrc.oss.js) file which is copied in to export package after changes in https://github.com/databricks/universe/pull/400945
I'll just remove those blocks, it seem to work

Comment (hubertzub-db) [reply mlflow/server/js/.eslintrc.js]: thanks for catching this! 🙏

**Pull request merged**